grpc.go

```diff
@@ -993,13 +993,31 @@ func (a *agentGRPC) RemoveContainer(ctx context.Context, req *pb.RemoveContainer
 		if err := ctr.removeContainer(); err != nil {
 			return emptyResp, err
 		}
+
+		// Find the sandbox storage used by this container
+		for _, path := range ctr.mounts {
+			if _, ok := a.sandbox.storages[path]; ok {
+				if err := a.sandbox.unsetAndRemoveSandboxStorage(path); err != nil {
+					return emptyResp, err
+				}
+			}
+		}
 	} else {
 		done := make(chan error)
 		go func() {
 			if err := ctr.removeContainer(); err != nil {
 				done <- err
```

Comment:
You need to return and close(done) here, or if error happens in later part, the done channel will block and go routine will leak

Reply:
Not part of my changes but the channel is getting closed a few lines below.

Reply:
Makes sense. Thanks.